microsoft/testsuites/xfstests/xfstests.py

```diff
@@ -171,11 +171,15 @@ def run_test(
 
         pgrep = self.node.tools[Pgrep]
         # this is the actual process name, when xfstests runs.
-        pgrep.wait_processes("check", timeout=timeout)
-
-        self.check_test_results(
-            log_path=log_path, test_type=test_type, result=result, data_disk=data_disk
-        )
+        try:
+            pgrep.wait_processes("check", timeout=timeout)
+        finally:
+            self.check_test_results(
```

Comment:
Make sure the `check_test_results` won't raise another exception by incomplete content.